core/templates/components/graph-services/graph-layout.service.ts

```diff
@@ -520,7 +521,9 @@ export class StateGraphLayoutService {
   }
 
   getAugmentedLinks(
-      nodeData: NodeDataDict, nodeLinks: GraphLink[]): AugmentedLink[] {
+      nodeData: NodeDataDict,
+      nodeLinks: GraphLink[]
+  ): AugmentedLink[] | undefined {
```

Comment:
Why can this return `undefined`?

Reply:
@vojtechjelinek  Here, we are not returning anything. **Refer lines: 544-547**

![image](https://user-images.githubusercontent.com/75239620/144302490-58430bc9-fbdd-4f2e-8092-30b21f8168d0.png)

Also, **Lines: 485-527** inside **.spec.ts** file also have something to do with this. We are expecting it to be undefined when 
`State1.xLabel === State2.xLabel` and `State1.yLabel === State2.yLabel`
![image](https://user-images.githubusercontent.com/75239620/144303318-11477eeb-80e5-4237-853e-cf7eed80f1c0.png)

Reply:
Okay, please add an explanatory code comment.

Reply:
Done.